src/main/java/com/mparticle/kits/GoogleAnalyticsFirebaseKit.java

```diff
@@ -233,17 +233,21 @@ PickyBundle getCommerceEventBundle(CommerceEvent commerceEvent) {
         // each of these has an extra parameter that is optional
         Map<String, List<String>> customFlags = commerceEvent.getCustomFlags();
         if (customFlags != null && customFlags.containsKey(CF_GA4COMMERCE_EVENT_TYPE)) {
-            // TODO: Will get 0 throw an error if the array is empty?
-            String commerceEventType = customFlags.get(CF_GA4COMMERCE_EVENT_TYPE).get(0);
-            if (commerceEventType.equals(FirebaseAnalytics.Event.ADD_SHIPPING_INFO.toString())) {
-                if (customFlags.containsKey(CF_GA4_SHIPPING_TIER)) {
-                    pickyBundle.putString(FirebaseAnalytics.Param.SHIPPING_TIER, customFlags.get(CF_GA4_SHIPPING_TIER).get(0));
-                }
-            } else if (commerceEventType.equals(FirebaseAnalytics.Event.ADD_PAYMENT_INFO.toString())) {
-                if (customFlags.containsKey(CF_GA4_PAYMENT_TYPE)) {
-                    pickyBundle.putString(FirebaseAnalytics.Param.PAYMENT_TYPE, customFlags.get(CF_GA4_PAYMENT_TYPE).get(0));
+            List<String> commerceEventTypeList = customFlags.get(CF_GA4COMMERCE_EVENT_TYPE);
+                if (commerceEventTypeList != null && commerceEventTypeList.size() > 0) {
```

Comment:
Not sure what happened here, but looks like lines 237 - 249 are indented one level too far.

Reply:
Pushed a commit to fix it.